lib/fizzy/execute.cpp

```diff
@@ -83,6 +83,28 @@ std::vector<TypeIdx> match_imports(const Module& module,
     return imported_function_types;
 }
 
+uint64_t eval_constant_expression(ConstantExpression expr,
+    const std::vector<ImportedGlobal>& imported_globals, const std::vector<Global>& global_types,
+    const std::vector<uint64_t>& globals)
+{
+    if (expr.kind == ConstantExpression::Kind::Constant)
+        return expr.value.constant;
+
+    assert(expr.kind == ConstantExpression::Kind::GlobalGet);
+
+    const auto global_idx = expr.value.global_index;
+    const bool is_mutable = (global_idx < imported_globals.size() ?
+                                 imported_globals[global_idx].is_mutable :
+                                 global_types[global_idx - imported_globals.size()].is_mutable);
+    if (is_mutable)
```

Comment:
This validation should perhaps be done in parser (constant expression is invalid if it uses mutable globals) . But currently it seems much easier to do it in instantiate.